### ISSUE TITLE
Not sure what's wrong with this one.

### ISSUE THREAD

**Issue by menghan465** [statsmodels/statsmodels#9436]
import numpy as np
import pandas as pd
import statsmodels.api as sm
np.random.seed(123)
nobs = 1000
data = np.random.randn(nobs, 2)
df = pd.DataFrame(data, columns=['y1', 'y2'])
A = np.array([[1, 0], [0.5, 1]])
B = np.identity(2)
svar_model = sm.tsa.SVAR(df, svar_type='A', A=A, B=B)
svar_result = svar_model.fit()
print(svar_result.summary())
<img width="682" alt="1732464486237" src="https://github.com/user-attachments/assets/4c7eefdf-a566-4bfa-9fe3-f3143bfccf1c">

**Comment by josef-pkt** [comment]
my guess is that it is the same as #9302

**Comment by josef-pkt** [comment]
mabye not

A and B are fully specified, there are no structural params to estimate

But A_mask and B_mask are boolean scalar instead of array.  
Something is wrong with shape if a mask is empty.

```
ipdb> p A
array([[1. , 0. ],
       [0.5, 1. ]])
ipdb> p A_mask
False
ipdb> start_params[:A_len]
array([], dtype=float64)
ipdb> start_params[A_len:]
array([], dtype=float64)
ipdb> start_params
array([], dtype=float64)
ipdb> B_mask
False
```

**Comment by josef-pkt** [comment]
@menghan465  What's your numpy version?

reason that maks is scalar boolean if empty is: 
```
ipdb> np.logical_or(A == 'E', A == 'e')
<stdin>:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
False
ipdb> np.__version__
'1.21.5'
```

**Comment by menghan465** [comment]
> @menghan465 What's your numpy version?
> 
> reason that maks is scalar boolean if empty is:
> 
> ```
> ipdb> np.logical_or(A == 'E', A == 'e')
> <stdin>:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
> False
> ipdb> np.__version__
> '1.21.5'
> ```

<img width="506" alt="微信图片_20241125011500" src="https://github.com/user-attachments/assets/a5eb7383-8a8c-443c-99a0-716918953743">

**Comment by josef-pkt** [comment]
ok, I will fix logical_or for, I guess, numpy < 2.0

However, that will not fix possible problems if both A and B have no parameters to estimate.
I don't remember how SVAR estimation works, but most likely it cannot handle empty parameters.


Another issue with my current numpy

dtype of A and B

A in example here is float/float64.
A in unit test has dtype "U"
internally stored, attributes A and B are float with np.nan instead of "E". 

```
(A == 'e')
...\Local\Temp\ipykernel_22988\2041847108.py:1: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  (A == 'e')
False

A.astype("U") == "e"
array([[False, False],
       [False, False]])
```
